### ISSUE TITLE
GCop 621

### ISSUE THREAD

**Issue by marjanjavid** [Geeksltd/GCop#93]
I get Gcop621 error on this line of my code:
private static byte[] CreateBytes(int count = 100) => Enumerable.Range(0, count).Select(i => (byte)i).ToArray();

so i changed it to the below:
private static byte[] CreateBytes(int count = 100) => Enumerable.Range(0, count).Cast<byte>().ToArray();

when i use cast() i get "Specified cast is not valid." error.

**Comment by PaymonK** [comment]
This rule should be skipped if there is an [explicit cast operator defined](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/explicit) from the type of `T` (of the list object) and the type of the destination.
 
Note: In the above situation, the explicit cast invocation is not the same as calling the Linq Cast() method.

**Comment by Foroughi** [comment]
@PaymonK I dont follow you. By any chance did you mean : 

This rule **should kick in** only if there is an explicit cast operator defined from the type of T (of the list object) and the type of the destination.

**Comment by PaymonK** [comment]
I see.

Perhaps for now, we can just hard code the exception for the casts that we know are problematic, e.g. (i => (byte)i).

**Comment by sungam3r** [comment]
```csharp
         private static object SomeTransform(object obj) => obj.ToString();

        private static IEnumerable<string> Get(this IEnumerable<object> collection)
        {
            return collection.Select(item => (string)SomeTransform(item)); // GCop621	Use Cast<string>() here instead.
        }
```

But `Cast<string>` is not applicable here.
